examples/callbacks/README.md

```diff
@@ -0,0 +1,140 @@
+
+| Callback      | When Triggered                  | Purpose                                       |
+| ------------- | ------------------------------- | --------------------------------------------- |
+| `BeforeAgent` | Before agent execution starts   | Guardrails, validation, early returns         |
+| `AfterAgent`  | After agent execution completes | Post-processing, logging, output modification |
+| `BeforeModel` | Before each LLM call            | Caching, request modification, guardrails     |
+| `AfterModel`  | After each LLM response         | Response modification, logging, sanitization  |
+| `BeforeTool`  | Before each tool execution      | Authorization, caching, argument modification |
+| `AfterTool`   | After each tool execution       | Result modification, logging, sanitization    |
+
+## Flow Control
+
+- **Before callbacks**: Return `nil` to continue normal execution, or return a value to skip the default behavior and use the returned value instead.
+- **After callbacks**: Return `nil` to use the original output, or return a modified value to replace it.
+
+## Usage Example
+
+```go
+callbackConfig := &server.CallbackConfig{
+    BeforeAgent: []server.BeforeAgentCallback{
+        func(ctx context.Context, callbackCtx *server.CallbackContext) *types.Message {
+            // Return nil to proceed, or return a message to skip agent execution
+            return nil
+        },
+    },
+    AfterAgent: []server.AfterAgentCallback{
+        func(ctx context.Context, callbackCtx *server.CallbackContext, output *types.Message) *types.Message {
+            // Return nil to use original output, or return modified output
+            return nil
+        },
+    },
+    BeforeModel: []server.BeforeModelCallback{
+        func(ctx context.Context, callbackCtx *server.CallbackContext, request *server.LLMRequest) *server.LLMResponse {
+            // Return nil to proceed with LLM call, or return a response to skip it
+            return nil
+        },
+    },
+    BeforeTool: []server.BeforeToolCallback{
+        func(ctx context.Context, tool server.Tool, args map[string]interface{}, toolCtx *server.ToolContext) map[string]interface{} {
+            // Return nil to execute tool, or return a result to skip execution
+            return nil
+        },
+    },
+}
+
+agent, err := server.NewAgentBuilder(logger).
+    WithConfig(&cfg.AgentConfig).
+    WithLLMClient(llmClient).
+    WithCallbacks(callbackConfig).
+    Build()
+```
+
+## Use Cases
+
+### Guardrails
+
+Use `BeforeAgent` or `BeforeModel` to block or modify requests based on content policies.
+
+### Caching
+
+Use `BeforeModel` to return cached responses and skip expensive LLM calls.
+
+### Authorization
+
+Use `BeforeTool` to implement tool-level authorization checks.
+
+### Logging & Metrics
+
+Use any callback to log execution details or send metrics.
+
+### Response Sanitization
+
+Use `AfterModel` or `AfterTool` to remove sensitive information from outputs.
+
```

Comment:
This just repeats the same info in the table, either provide a small example or delete